index.js

```diff
@@ -5,8 +5,31 @@
  * MIT Licensed
  */
 var promiseExists = typeof Promise === 'function';
-var globalObject = typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : self; // eslint-disable-line
-var isDom = 'location' in globalObject && 'document' in globalObject;
+
+/* eslint-disable */
+// See http://stackoverflow.com/a/6930376
+var globalObject;
+try {
+  globalObject = Function('return this')() || (42, eval)('this');
```

Comment:
This was, for the most part, copied from that StackOverflow post. The `try/catch` is needed because of CSP. `window` is a safe fallback because CSP doesn't exist in non-browser environments, afaik. The `||` I wasn't quite as clear on. Something to do with `use strict` if I was reading that SO post correctly?

@shvaikalesh will it "not work" even with the `try/catch`? I thought I was already accounting for CSP... Also, I agree about using `new`.